includes/Context.php

```diff
@@ -140,6 +140,37 @@ public function get_reference_site_url() {
 		return $site_url;
 	}
 
+	/**
+	 * Gets the permalink of the reference site to use for stats.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param int|\WP_Post $post  Optional. Post ID or post object. Default is the global `$post`.
+	 *
+	 * @return string|false The  Reference permalink URL or false if post does not exist.
+	 */
+	public function get_reference_permalink( $post = 0 ) {
+		$reference_site_url = untrailingslashit( $this->get_reference_site_url() );
+
+		// Returns home page url when front page mode is latest blog posts.
+		if ( ! $post && 'posts' === get_option( 'show_on_front' ) && is_home() ) {
+			return trailingslashit( $reference_site_url );
+		}
```

Comment:
As discussed yesterday, this condition shouldn't exist. For now, we should only cover single posts. Covering any kind of archive is an enhancement for the future.

Reply:
Instead, the same logic that you have in the `Admin_Bar` class (conditional usage of `get_queried_object()` vs `get_post()`) should be used here.

Reply:
Address in https://github.com/google/site-kit-wp/pull/73/commits/149af340e745c69f727713b5f99892abc94737d3